src/crud.test.js

```diff
@@ -0,0 +1,54 @@
+import { addToArr, removeFromArr } from './crud.js';
+
+const arr = [1];
+
+describe('Test add to array function', () => {
+  test('Adding a new element at 1', () => {
+    addToArr('newTask1', arr);
+    expect(arr).toHaveLength(2);
+  });
+
+  test('Adding a new element at 2', () => {
+    addToArr('newTask2', arr);
+    expect(arr).toHaveLength(3);
+  });
+
+  test('Adding a new element at 3', () => {
+    addToArr('newTask3', arr);
+    expect(arr[3].index).toBe(4);
+  });
+
+  test('Adding a new element at 4', () => {
+    addToArr('New text', arr);
+    expect(arr[4].description).toBe('New text');
+  });
+  test('Adding a new element at 5', () => {
+    addToArr('newTask5', arr);
```

Comment:
Hi there 👋🏾  your test looks pretty good and detailed but as required you  still need to create a mock  of the local storage, DOM or External API of your web app

Reply:
Hi @alexander16108 and thank you for the feedback.
As you know this is **Testing To Do list: part 1** and the requirement is to write tests for the "add" and "remove" functions.

Here in the [requirements ](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_1.md)says that:
> Take a look at each of these functions. Are they pure functions? If the answer is "yes" then writing tests for them should be straightforward. Some of those functions however will update localStorage and manipulate the DOM. For those, you will need to use mocks

So as our "add" and "remove" functions were pure functions we wrote the tests like that.
Obviously in the **Testing To Do list: part 2** we have to write mock tests of the local storage, DOM.

Please correct me if I'm wrong.
Thanks